app/services/markdown_processor/parser.rb

```diff
@@ -28,12 +28,16 @@ def finalize(link_attributes: {})
       begin
         liquid_tag_options = { source: @source, user: @user }
 
+        # check value of sanitized_content to see if the .to_str method removes spaces
+
         # NOTE: [@rhymes] liquid 5.0.0 does not support ActiveSupport::SafeBuffer,
         # a String substitute, hence we force the conversion before passing it to Liquid::Template.
         # See <https://github.com/Shopify/liquid/issues/1390>
         parsed_liquid = Liquid::Template.parse(sanitized_content.to_str, liquid_tag_options)
-
-        html = markdown.render(parsed_liquid.render)
+        options2 = { hard_wrap: false, filter_html: false, link_attributes: link_attributes }
```

Comment:
Since the lines you've added only depend on the local variables `parsed_liquid` and `link_attributes`, it might make sense to move this second pass to a separate method with a name to indicate this markdown render is intended to strip unneeded line breaks. That approach also allows for cleaning up the variable names since they're no longer conflicting/shadowing.  Would something like this work? 

```ruby
def remove_extra_linebreaks(parsed_liquid, link_attributes)
  options = { hard_wrap: false, filter_html: false, link_attributes: link_attributes)
  renderer = Redcarpet::Render::HTMLRouge.new(options)
  markdown = Redcarpet::Markdown.new(renderer, Constants::Redcarpet::CONFIG)
  markdown.render(parsed_liquid.render)
end
```

Reply:
I think this would work - I wasn't sure what to name this before I had to drop this issue, but this sounds good to me.

Reply:
Thanks for the feedback - and I understand you're now busier than you were last month. I'll putter around the edges seeing what I can do, and will probably push up a commit later today, I'm not sold on this idea and it probably misses some better idea. 

I'm more interested in the inconsistent rendering of the `~~~` blocks I noticed when trying to confirm the change had the intended effect (earlier comment), and this issue and bugfix are incidental to that.